src/chmatch.c

```diff
@@ -52,19 +45,18 @@ static SEXP chmatchMain(SEXP x, SEXP table, int nomatch, bool chin, bool chmatch
     SET_TRUELENGTH(s,0);   // TODO: do we need to set to zero first (we can rely on R 3.1.0 now)?
   }
   const int tablelen = length(table);
-  const SEXP *td = STRING_PTR(table);
+  SEXP utable = table;
+  for (int i=0; i<tablelen; i++) {
```

Comment:
this seems quite useful, is it something that can be functionalized & put into utils.c for easy access elsewhere?

coerceUtfIfNeeded or something

Reply:
Good idea. I've functionalized them.